vision_agent/tools/tools.py

```diff
@@ -1746,6 +1746,107 @@ def closest_box_distance(
     return cast(float, np.sqrt(horizontal_distance**2 + vertical_distance**2))
 
 
+def flux_image_inpainting(
+    prompt: str,
+    image: np.ndarray,
+    mask_image: np.ndarray,
+    height: Optional[int] = None,
+    width: Optional[int] = None,
+) -> List[np.ndarray]:
+    """'flux_image_inpainting' performs image inpainting using a rectified flow transformer model to
+    fill masked regions of an image based on textual prompts and surrounding context.
+    The function uses image generation model and transformer architecture with rectified flows to
+    understand both the textual prompt and visual context, ensuring the inpainted regions
+    blend naturally with the preserved areas.
+
+    Parameters:
+        prompt (str):
+            A detailed text description guiding what should be generated in the masked area.
+            More detailed and specific prompts typically yield better results.
+        image (np.ndarray):
+            The source image to be inpainted. The image will serve as the base context for
+            the inpainting process.
+        mask_image (np.ndarray):
+            A binary mask image where:
+            - White pixels (255) indicate areas to be inpainted/replaced
+            - Black pixels (0) indicate areas to be preserved
+            The mask should have the same dimensions as the input image.
```

Comment:
Change this to handle the same mask structure that's output by other functions like `florence2_sam2_image` which is a numpy array of 0's and 1's. Then convert it internally, this way the LLM doesn't have to do the conversion externally which can cause unnecessary debug cycles

Reply:
Resolved! Raising an error if the input mask is not binary